ir/src/constraints/graph.rs

```diff
@@ -472,6 +483,16 @@ impl AlgebraicGraph {
             )),
         }
     }
+
+    fn insert_random_value(
+        &mut self,
+        index: usize,
+        domain: ConstraintDomain,
+        trace_segment: u8,
+    ) -> Result<ExprDetails, SemanticError> {
+        let node_index = self.insert_op(Operation::RandomValue(index));
+        Ok(ExprDetails::new(node_index, trace_segment, domain))
+    }
```

Comment:
nit: I'd move this method to just below `insert_constant`. The ordering of the methods has "access" methods after the others.